nanomq/apps/broker.c

```diff
@@ -486,7 +441,10 @@ server_cb(void *arg)
 				nng_aio_finish(work->aio, 0);
 				break;
 			} else {
-				if (smsg) {
+				// TODO Do some optimization
+				// Free the msg with qos 0 (connackNotify actually)
+				// But not free the msg from nano_qos_db
+				if (smsg && 0 == work->pub_packet->fixed_header.qos) {
```

Comment:
Please keep msg ref count clean. Make ref count ++ each time when store msg in cache_session